userland/src/parameters.cpp

```diff
@@ -0,0 +1,74 @@
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <iostream>
+#include <string>
+#include <thread>
+
+#include <rclcpp/rclcpp.hpp>
+
+template <typename ParameterTypeT>
+ParameterTypeT
+wait_for_future(rclcpp::node::Node::SharedPtr & node, std::shared_future<ParameterTypeT> & future)
+{
+  std::future_status status;
+  do {
+    rclcpp::spin_some(node);
+    status = future.wait_for(std::chrono::milliseconds(100));
+  } while (status != std::future_status::ready && rclcpp::ok());
+
+  if (std::future_status::ready == status) {
+    return future.get();
+  }
+
+  throw std::runtime_error("Invalid future status");
+}
```

Comment:
I meant hiding this code inside the synchronous calls, so that a synchronous client would return the value held by the future directly.

I'm ok with exposing `wait_for_future` somehow, but not extending `std::future`